src/wordlift/mappings/class-mappings-dbo.php

```diff
@@ -271,20 +295,30 @@ public function get_rule_groups_by_mapping( $mapping_id ) {
 	/**
 	 * Gets a list of rule items belong to rule_group_id.
 	 *
-	 * @param int $rule_group_id Indicates which group the item belongs to.
+	 * @param string $rule_group_id Indicates which group the item belongs to.
 	 *
 	 * @return array Get list of rule items.
 	 */
 	public function get_rules_by_rule_group( $rule_group_id ) {
-		$rule_table_name = $this->wpdb->prefix . WL_RULE_TABLE_NAME;
+		$rule_table_name  = $this->wpdb->prefix . WL_RULE_TABLE_NAME;
+		$rule_group_id    = (string) $rule_group_id;
+		$ui_defined_rules = array();
+		if ( is_numeric( $rule_group_id ) ) {
```

Comment:
@naveen17797 pls add comments to explain what's going on here: why we may have non numeric ids?